src/scripting/lua_widget_attributes.cpp

```diff
@@ -0,0 +1,527 @@
+{
+	w.select_page(value -1);
+}
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	return w.current_layer() + 1;
+}
+
+WIDGET_SETTER("value_compat,selected_index", int, gui2::stacked_widget)
+{
+	w.select_layer(value - 1);
+}
+
+WIDGET_GETTER("selected_index", int, gui2::selectable_item)
+{
+	return w.get_value() + 1;
+}
+
+WIDGET_SETTER("selected_index", int, gui2::selectable_item)
+{
+	if(value > int(w.num_states())) {
+		throw std::invalid_argument("invalid index");
+	}
+	w.set_value(value + 1);
+}
+
+WIDGET_GETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	if(w.num_states() == 2) {
+		return w.get_value_bool();
+	}
+	throw std::invalid_argument("invalid widget");
+}
+
+WIDGET_SETTER("value_compat,selected", bool, gui2::selectable_item)
+{
+	w.set_value_bool(value);
+}
+
+WIDGET_GETTER("value_compat,text", std::string, gui2::text_box)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,text", std::string, gui2::text_box)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,value", int, gui2::slider)
+{
+	return w.get_value();
+}
+
+WIDGET_SETTER("value_compat,value", int, gui2::slider)
+{
+	w.set_value(value);
+}
+
+WIDGET_GETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	return w.get_percentage();
+}
+
+WIDGET_SETTER("value_compat,percentage", int, gui2::progress_bar)
+{
+	w.set_percentage(value);
+}
+
+WIDGET_GETTER("value_compat,selected_item_path", std::vector<int>, gui2::tree_view)
+{
+	auto res = w.selected_item()->describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_GETTER("path", std::vector<int>, gui2::tree_view_node)
+{
+	auto res = w.describe_path();
+	for(int& a : res) { ++a;}
+	return res;
+}
+
+WIDGET_SETTER("value_compat,unfolded", bool, gui2::tree_view_node)
+{
+	if(value) {
+		w.unfold();
+	} else {
+		w.fold();
+	}
+}
+
+WIDGET_SETTER("value_compat,unit", lua_index_raw, gui2::unit_preview_pane)
+{
+	if(const unit_type* ut = luaW_tounittype(L, value.index)) {
+		w.set_displayed_type(*ut);
+	} else if(unit* u = luaW_tounit(L, value.index)) {
+		w.set_displayed_unit(*u);
+	} else {
+		luaW_type_error(L, value.index, "unit or unit type");
+	}
+}
+
+WIDGET_GETTER("item_count", int, gui2::multi_page)
+{
+	return w.get_page_count();
+}
+
+WIDGET_SETTER("use_markup", bool, gui2::styled_widget)
+{
+	w.set_use_markup(value);
+}
+
+//TODO: while i think this shortcut is useful, i'm not that happy about
+//      the name since  it changes 'label' and not 'text', the first one
+//      is the label that is part of most widgets (like checkboxes), the
+//      later is sepcific to input textboxes.
+WIDGET_SETTER("marked_up_text", t_string, gui2::styled_widget)
+{
+	w.set_use_markup(true);
+	w.set_label(value);
+}
+
+WIDGET_SETTER("enabled", bool, gui2::styled_widget)
+{
+	w.set_active(value);
+}
+
+WIDGET_SETTER("tooltip", t_string, gui2::styled_widget)
+{
+	w.set_tooltip(value);
+}
+
+
+WIDGET_SETTER("callback", lua_index_raw, gui2::widget)
```

Comment:
I don't like calling this `callback`, because the actual meaning of it depends on the widget, doesn't it? I'd prefer descriptive names beginning with `on`, for example `onchange` or `onclick`.

Or is this a compatibility layer similar to `value_compat`? In that case, I can't help wondering why it doesn't also have `_compat` in its name... plus, if that's the case, you shouldn't've used it in the zombie dialog.

Reply:
> Or is this a compatibility layer similar to value_compat? In

I _think_ it is but i'm not 100% sure

Reply:
So, I notice it's used in the zombie dialog. Is there a way to not do that? Or do you perhaps need to add more `on_xxx` callbacks? The zombie dialog seems to be doing an "on select" event, not sure what the actual underlying event is though.

Reply:
i think the on_modifed event is fired in this case.